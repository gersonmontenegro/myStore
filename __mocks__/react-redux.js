/* eslint-disable arrow-parens */
/* eslint-disable no-unused-vars */
jest.mock('react-redux', () => ({
    // useSelector: jest.fn().mockReturnValue(props => null),
    useSelector: (d) => d,
    useDispatch: jest.fn(),
}));
