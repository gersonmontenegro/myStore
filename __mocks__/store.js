import configureStore from 'redux-mock-store';

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
    dataReducer: {
        data: 99,
    },
};

const store = mockStore(initialState);

export default store;
