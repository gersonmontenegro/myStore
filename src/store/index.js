/* eslint-disable import/no-unresolved */
import { createStore, compose } from 'redux';
import reactotron from 'initStack/ReactotronConfig';
import reducers from 'reducers';

export default createStore(
    reducers,
    compose(reactotron.createEnhancer()),
);
