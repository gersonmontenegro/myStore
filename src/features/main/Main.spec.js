/* eslint-disable import/no-unresolved */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import store from 'mocks/store';
import Main from './index';

describe('Test Main feature', () => {
    let wrapper;
    const props = {
        navigation: {
            navigate: jest.fn(),
        },
    };

    beforeEach(() => {
        wrapper = shallow(
            <Provider store={store}>
                <Main {...props} />
            </Provider>,
        );
    });

    test('Render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
