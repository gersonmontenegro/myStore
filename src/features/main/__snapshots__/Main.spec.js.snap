// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Main feature Render correctly 1`] = `ShallowWrapper {}`;
